tokenclient/tokenFlows.go

```diff
@@ -101,20 +119,25 @@ func (t *TokenFlows) ClientCredentials(ctx context.Context, customerTenantURL st
 	if err != nil {
 		return "", err
 	}
-	r, err := http.NewRequestWithContext(ctx, http.MethodPost, targetURL, strings.NewReader(data.Encode())) // URL-encoded payload
+	r, err := t.createRequestWithContext(ctx, targetURL, data) // URL-encoded payload
 	if err != nil {
 		return "", fmt.Errorf("error performing client credentials flow: %w", err)
 	}
-	r.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	r.httpRequest.Header.Set("Content-Type", "application/x-www-form-urlencoded")
 
-	var tokenRes tokenResponse
-	err = t.performRequest(r, &tokenRes)
+	return t.getOrRequestToken(r)
+}
+
+func (t *TokenFlows) createRequestWithContext(ctx context.Context, targetURL string, data url.Values) (request, error) {
+	r, err := http.NewRequestWithContext(ctx, http.MethodPost, targetURL, strings.NewReader(data.Encode())) // URL-encoded payload
 	if err != nil {
-		return "", err
-	} else if tokenRes.Token == "" {
-		return "", fmt.Errorf("error parsing requested client credentials token: no 'access_token' property provided")
+		return request{}, err
 	}
-	return tokenRes.Token, nil
+	return request{
+		targetURL:   targetURL,
+		parameters:  data.Encode(),
```

Comment:
nit: could be outlined (together with #L132) so that data only needs to be encoded once.